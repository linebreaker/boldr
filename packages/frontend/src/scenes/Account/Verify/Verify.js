/* @flow */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import Button from '@boldr/ui/Button';
// internal
import FormCard from '@boldr/ui/Form/FormCard/FormCard';
import { verifyAccount } from '../state/actions';

export type Props = {
  dispatch: () => void,
  match: Object,
};

class Verify extends Component {
  props: Props;

  handleVerify = (event: Event): void => {
    event.preventDefault();
    this.props.dispatch(verifyAccount(this.props.match.params.token));
  };

  render() {
    return (
      <div className="verify-wrapper">
        <Helmet title="Verify Account" />
        <FormCard
          title="Account verification"
          skinny
          lightText
          form={
            <Button onClick={this.handleVerify} kind="primary" block>
              Verify
            </Button>
          }
        />
      </div>
    );
  }
}

export default connect()(Verify);
