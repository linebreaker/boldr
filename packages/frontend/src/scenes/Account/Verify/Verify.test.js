import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import Verify from './Verify';

test('<Verify />, renders the verify account form card', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);
  const wrapper = shallow(
    <Provider store={store}>
      <Verify />
    </Provider>,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
