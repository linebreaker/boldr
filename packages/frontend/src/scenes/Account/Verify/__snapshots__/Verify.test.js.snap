// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Verify />, renders the verify account form card 1`] = `<Connect(Verify) />`;
